src/components/network-modification-tree-pane.js

```diff
@@ -88,6 +93,92 @@ export const NetworkModificationTreePane = ({
     const { snackError, snackInfo } = useSnackMessage();
     const classes = useStyles();
     const DownloadIframe = 'downloadIframe';
+    const [notifListenedStudyId, setNotifListenedStudyId] = useState(null);
+    const [ws, setWs] = useState(null);
+    const broadcastChannel = useSelector((state) => state.broadcastChannel);
+    const dispatchSelectedNodeForCopy = useCallback(
+        (sourceStudyId, nodeId, copyType) => {
+            dispatch(
+                setSelectedNodeForCopy({
+                    sourceStudyId: sourceStudyId,
+                    nodeId: nodeId,
+                    copyType: copyType,
+                })
+            );
+        },
+        [dispatch]
+    );
+
+    useEffect(() => {
+        //initialise channel onmessage
+        if (typeof broadcastChannel.onmessage !== 'function') {
+            broadcastChannel.onmessage = (event) => {
+                console.info('message received from broadcast channel');
+                console.info(event.data);
+                setNotifListenedStudyId(event.data.sourceStudyId);
+                dispatchSelectedNodeForCopy(
+                    event.data.sourceStudyId,
+                    event.data.nodeId,
+                    CopyType.COPY
+                );
+            };
+        }
+    });
+
+    useEffect(() => {
+        //The first time we initialise the websocket
+        if (ws === null && notifListenedStudyId !== null) {
+            const initWs = connectNotificationsWebsocket(notifListenedStudyId);
+            initWs.onmessage = function (event) {
+                //If the study we are on is the one that updated the node we don't need to handle it with this ws as it's already handled in a useEffect
+                if (event.data.sourceStudyId !== studyUuid) {
+                    let eventData = JSON.parse(event.data);
+                    if (
+                        eventData.headers &&
+                        eventData.headers.updateType === 'nodeUpdated'
+                    ) {
+                        if (
+                            eventData.headers.nodes.some(
+                                (nodeId) =>
+                                    nodeId ===
+                                    selectedNodeForCopyRef.current.nodeId
+                            )
+                        ) {
+                            dispatch(
+                                setSelectedNodeForCopy(noSelectionForCopy)
```

Comment:
unify with previous mechanism for invalidating copy

Reply:
done